app/internal/wm/os_android.go

```diff
@@ -609,6 +611,31 @@ func (w *window) ShowTextInput(show bool) {
 	})
 }
 
+func (w *window) SetInputHint(mode key.InputHint) {
+	// Constants defined at https://developer.android.com/reference/android/text/InputType.
+	const (
+		TYPE_NULL                = 0
+		TYPE_CLASS_NUMBER        = 2
+		TYPE_NUMBER_FLAG_DECIMAL = 8192
+		TYPE_NUMBER_FLAG_SIGNED  = 4096
+	)
+
+	runOnMain(func(env *C.JNIEnv) {
```

Comment:
SetInputInput should be called through Window.Run and you can assume the main thread here. See window.ShowTextInput in this file.